docs/DeveloperGuide.md

```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+2. Enter `add n/ab p/12345678` in the command box.
+
+	Expected: SociaLite shows error message "A contact with the same phone number already exists in the contact list!".
+
+3. Enter `add n/aa` in the command box.
+
+	Expected: SociaLite shows error message, indicating invalid command format given. This is because name and phone fields are compulsory for the addition of contact.
+
+4. Enter `add n/aa p/87654321` in the command box.
+
+	Expected: A new contact card with name "aa" and phone number "87654321" is created. SociaLite allows repetitive names in the contact list, but uses phone number as the unique identifier for a contact.
+
+5. Enter `add n/John Appleseed p/13572468 fb/john.appleseed twitter/morethanfifteenchars`.
+
+	Expected: SociaLite shows error message, indicating that the telegram username does not follow their naming convention.
+
+### Editing a contact
+
+Assuming all test cases in the previous section were carried out, and that the first contact card in the list is "aa", and there are only 10 contacts stored in SociaLite:
+
+1. Enter `edit 1 n/abc` in the command box.
+
+	Expected: the first contact card's name changes to "abc". It might not be the first contact card in the list due to alphabetical ordering.
+
+2. Enter `edit 100 n/abc` in the command box.
+
+	Expected: SociaLite shows error message, indicating invalid person index provided.
+
+3. Enter `edit 2` in the command box.
+
+	Expected: SociaLite shows error message, indicating at least one field to edit must be provided.
+
+4. Enter `edit 1 fb/hello` in the command box.
+
+	Expected: the first contact card in SociaLite should have a Facebook icon showed up at the bottom, with its handle being "hello".
+
+5. Enter `edit 1 t/tag1 t/tag2 t/tag3 t/tag4` followed by `edit 1 t/` in the command box.
+
+	Expected: the first contact card in SociaLite will be shown to have four tags, and all four tags will disappear upon entering the second command.
+
+### Deleting a contact
+
+Assuming there are 10 contacts stored in SociaLite:
+
+1. Enter `delete 1` in the command box.
+
+	Expected: the first contact card in SociaLite will be deleted. There will be 9 contacts in SociaLite.
+
+2. Enter `delete 100` in the command box.
+
+	Expected: SociaLite shows error message, indicating invalid person index provided.
+
+3. Enter `delete 0` in the command box.
+
+	Expected: SociaLite shows error message, indicating index must be non-zero unsigned integer.
+
+### Finding a contact
+
+Assuming there is a contact called "Aaron Tan" in SociaLite:
+
+1. Enter `find aaron tan` in the command box.
+
+	Expected: only contact cards with the name "aaron tan" (case-insensitive) will show up in the list.
+
+2. Enter `find a t` in the command box.
+
+	Expected: all contacts whose name starts with "a" and "t" (e.g. Aaron Tan, Alice Teo, Arthur Taylor) will show up in the list.
```

Comment:
Sorry! I'm just suggesting this change for greater clarity, in case the reader assumes that names like "Tony Ang" and "Terry Au" will be displayed as well.

"Expected: all contacts whose name contain two or more words, with the first word starting with 'a' and the second word starting with 't' (e.g. Aaron Tan, Alice Teo, Arthur Taylor) will show up in the list."

Reply:
Actually for this one I believe the implementation will actually show both "Aaron Tan" and "Tony Ang"! But this is a great point, I will include this in the DG as well!